Library/Homebrew/diagnostic.rb

```diff
@@ -884,9 +884,17 @@ def check_deleted_formula
         kegs = Keg.all
 
         deleted_formulae = kegs.map do |keg|
-          next if !CoreTap.instance.installed? && EnvConfig.install_from_api? && Tab.for_keg(keg).tap.core_tap?
+          next if Formulary.tap_paths(keg.name).any?
+
+          if !CoreTap.instance.installed? && EnvConfig.install_from_api?
+            # Formulae installed with HOMEBREW_INSTALL_FROM_API should not count as deleted formulae
+            # but may not have a tap listed in their tab
+            tap = Tab.for_keg(keg).tap
+            next if tap.present? && tap.core_tap?
+            next if tap.blank? && Homebrew::API::Bottle.available?(keg.name)
```

Comment:
I think this makes more sense?

```suggestion
            next if (tap.blank? || tap.core_tap?) && Homebrew::API::Bottle.available?(keg.name)
```

Reply:
Yeah, I like that. Thanks, @Bo98!